### PR TITLE
MINOR: clean up and add getters to config

### PR DESCRIPTION
**Pull request by levzem**
Signed-off-by: Lev Zemlyanov <lev@confluent.io>

## Problem
connector does not use config getters which is not the best practice and ends up with some clunky code

## Solution
add config getters and reorganize config class

##### Does this solution apply anywhere else?
- [ ] yes
- [x] no

## Test Strategy
ensure all previous tests pass

##### Testing done:
- [x] Unit tests
- [x] Integration tests
- [ ] System tests
- [ ] Manual tests

## Release Plan

 
backport to `5.0.x` to reduce pint merge conflicts in future PRs

### CODE REVIEW

File: src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java [hunk 1]
```diff
@@ -16,44 +16,67 @@
...
+  private static final String CONNECTION_USERNAME_DISPLAY = "Connection Username";
+  private static final String CONNECTION_USERNAME_DEFAULT = null;
+
   public static final String CONNECTION_PASSWORD_CONFIG = "connection.password";
   private static final String CONNECTION_PASSWORD_DOC =
       "The password used to authenticate with Elasticsearch. "
       + "The default is the null, and authentication will only be performed if "
       + " both the username and password are non-null.";
+  private static final String CONNECTION_PASSWORD_DISPLAY = "Connection Password";
+  private static final String CONNECTION_PASSWORD_DEFAULT = null;
```
Comment (Prathibha-m) [inline src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:58]: Should this have a default? or should it be `Importance.HIGH` and a required config?

File: src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java [hunk 2]
```diff
@@ -162,244 +231,245 @@ protected static ConfigDef baseConfigDef() {
...
+            Importance.MEDIUM,
+            CONNECTION_USERNAME_DOC,
+            CONNECTOR_GROUP,
+            ++order,
+            Width.SHORT,
+            CONNECTION_USERNAME_DISPLAY
+        ).define(
+            CONNECTION_PASSWORD_CONFIG,
+            Type.PASSWORD,
+            CONNECTION_PASSWORD_DEFAULT,
```
Comment (Prathibha-m) [inline src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:258]: Should this have a default? or should it be defined with `Importance.HIGH` and a required config?

Comment (levzem) [reply src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:258]: this is not a required config

File: src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java [hunk 3]
```diff
@@ -408,6 +478,117 @@ public ElasticsearchSinkConnectorConfig(Map<String, String> props) {
...
+  @Deprecated
+  public Map<String, String> topicToIndexMap() {
+    return parseMapConfig(getList(TOPIC_INDEX_MAP_CONFIG));
+  }
+
+  public String type() {
+    return getString(TYPE_NAME_CONFIG);
+  }
+
+  public Set<String> urls() {
```
Comment (dosvath) [inline src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:568]: It would be great if we could use a more descriptive name for this getter, ex. `connectionUrls`. Let's also make sure `getList` doesn't eliminate duplicates already.

File: src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java [hunk 4]
```diff
@@ -408,6 +478,117 @@ public ElasticsearchSinkConnectorConfig(Map<String, String> props) {
...
+
+  public String username() {
+    return getString(CONNECTION_USERNAME_CONFIG);
+  }
+
+  public boolean useCompactMapEntries() {
+    return getBoolean(COMPACT_MAP_ENTRIES_CONFIG);
+  }
+
+  private Map<String, String> parseMapConfig(List<String> values) {
```
Comment (dosvath) [inline src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:580]: What is this used for in this repo? First time I see this logic in connector configs.

**Review by dosvath — COMMENTED** [review]
Thanks for the cleanup @levzem! Few questions. 

One note for ES as a future improvement would be more integration test coverage around edge cases/negative paths, it seems to be behind compared to some of our other connectors.

File: src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java (hunk 3 above)
Comment (levzem) [reply src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:568]: it doesn't

File: src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java (hunk 4 above)
Comment (levzem) [reply src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java:580]: `TOPIC_INDEX_MAP_CONFIG`

**Review by dosvath — APPROVED** [review]
LGTM
